### DISCUSSION TITLE
Java/Scala usability

### DISCUSSION THREAD

**Discussion by jotterbach** [Z3Prover/z3#5026] [Ideas]
Hi folks,
I wondered if there are plans to make regular releases of *.jars to MavenCentral for use in Java/Scala projects. Also how do the jars reference the dynamic libraries? Do they need to be deployed somewhere, can they be bundled up in the jar?

**Comment by NikolajBjorner** [comment]
There is no support (from me) for various installers and distribution packages, but anybody is welcome to provide bridges either externally or by contributing pull requests to enable these bridges.

**Comment by wintersteiger** [comment]
Yes, we decided not to publish any packages in package repositories, because we couldn't possible maintain all of them by ourselves. But, you're welcome to do so for your favorite package manager!

I never looked into packaging the libraries into the jar as it would probably be way to much effort to produce one .jar that will run on all platforms. If you or someone else would like to take a crack at that
(and maintain it), that would of course be most welcome too.

**Comment by jotterbach** [comment]
Thank you @NikolajBjorner and @wintersteiger ! I might have a look at how to package up the libraries into a jar, but will probably stick with the python API for now then and only play with Java/Scala.

Just for completeness, I became aware of this project that seems to have a fairly comprehensive Scala API: https://github.com/epfl-lara/ScalaZ3
